source/features/fault.cc

```diff
@@ -325,6 +326,55 @@ namespace WorldBuilder
           total_fault_length[i] = local_total_fault_length;
           maximum_total_fault_length = std::max(maximum_total_fault_length, local_total_fault_length);
         }
+
+
+      std::vector<double> x_list(original_number_of_coordinates,0.0);
+      std::vector<double> y_list(original_number_of_coordinates,0.0);
+
+      for (size_t j=0; j<original_number_of_coordinates; ++j)
+        {
+          x_list[j] = coordinates[j][0];
+          y_list[j] = coordinates[j][1];
+        }
+
+      // Here, we compute the spherical bounding box using the two extreme points of the box containing all the surface
+      // coordinates and an additional buffer zone that accounts for the fault thickness and length. The first and second
+      // points correspond to the lower left and the upper right corners of the bounding box, respectively (see the
+      // documentation in include/bounding_box.h).
+      // For the spherical system, the buffer zone along the longitudal direction is calculated using the
+      // correponding latitude points.
+
+      const double radius_of_earth = 6371000.;
```

Comment:
let's talk about this. I don't like to have a constant in here. It would have my preference to have this at run time. I am curious how much of a performance difference it would make, since we can precompute everything except the radius of the earth. The other option would be to put it in the input file. There would be two options for that, either put it with the spherical coordinates or put it in a new section which contains optimization values. The downside of putting it in the spherical coordinates section is that it may conflict with the values provided by aspect and may confuse users. With a new section, we could make it clear that this should only be used for optimization and that it is the users responsibility to make sure it is consistent with the input file.

Reply:
Thank you for the reviews and your help! :) I pushed the changes in the latest commit (described more in the comment below)